test/widgets/text_test.dart

```diff
@@ -207,4 +206,65 @@ void main() {
     check(clampVariableFontWeight(999))  .equals(FontWeight.w900);
     check(clampVariableFontWeight(1000)) .equals(FontWeight.w900);
   });
+
+  group('proportionalLetterSpacing', () {
+    Future<void> testLetterSpacing(
+      String description, {
+      required double Function(BuildContext context) getValue,
+      double? ambientTextScaleFactor,
+      required double expected,
+    }) async {
+      testWidgets(description, (WidgetTester tester) async {
+        if (ambientTextScaleFactor != null) {
+          tester.platformDispatcher.textScaleFactorTestValue = ambientTextScaleFactor;
+          addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
+        }
+        await tester.pumpWidget(
+          MaterialApp(
+            home: Builder(builder: (context) => Text('',
+              style: TextStyle(letterSpacing: getValue(context))))));
+
+        final TextStyle? style = tester.widget<Text>(find.byType(Text)).style;
+        final actualLetterSpacing = style!.letterSpacing!;
+        check((actualLetterSpacing - expected).abs()).isLessThan(0.0001);
+      });
+    }
+
+    // From trying the options on an iPhone 13 Pro running iOS 16.6.1:
+    const textScaleFactors = <double>[
```

Comment:
In the tests, the `textScaleFactors` list is copied from
    emoji_reaction_test. Perhaps there's a convenient place to define it
    centrally.

I think probably most convenient would be to put it at the top level of this file `text_test.dart`, and then `emoji_reaction_test.dart` can import from there.

It feels slightly messy to have one file be importing from another file that has its own `main` function; but it works fine, and it makes a low-fuss solution here.